ecs/context.go

```diff
@@ -166,78 +282,69 @@ func (h contextCreateAWSHelper) chooseProfile(profiles []string) (string, error)
 		return "", err
 	}
 	profile := options[selected]
-	if options[selected] == "new profile" {
-		suggestion := ""
-		if !contains(profiles, "default") {
-			suggestion = "default"
-		}
-		name, err := h.user.Input("profile name", suggestion)
-		if err != nil {
-			return "", err
-		}
-		if name == "" {
-			return "", fmt.Errorf("profile name cannot be empty")
-		}
-		return name, h.createProfile(name)
-	}
 	return profile, nil
 }
 
-func (h contextCreateAWSHelper) chooseRegion(region string, profile string) (string, error) {
-	suggestion := region
-
+func getRegion(profile string) (string, error) {
+	if profile == "" {
+		profile = "default"
+	}
 	// only load ~/.aws/config
 	awsConfig := defaults.SharedConfigFilename()
 	configIni, err := ini.Load(awsConfig)
-
 	if err != nil {
 		if !os.IsNotExist(err) {
 			return "", err
 		}
 		configIni = ini.Empty()
 	}
+
+	getProfileRegion := func(p string) string {
+		r := ""
+		section, err := configIni.GetSection(p)
+		if err == nil {
+			reg, err := section.GetKey("region")
+			if err == nil {
+				r = reg.Value()
+			}
+		}
+		return r
+	}
 	if profile != "default" {
 		profile = fmt.Sprintf("profile %s", profile)
 	}
-	section, err := configIni.GetSection(profile)
-	if err != nil {
-		if !strings.Contains(err.Error(), "does not exist") {
-			return "", err
+	region := getProfileRegion(profile)
+	if region == "" {
+		region = getProfileRegion("default")
+		if region == "" {
+			return "us-east-1", nil
 		}
-		section, err = configIni.NewSection(profile)
+		return region, nil
+	}
+	return region, nil
+}
+
+func (h contextCreateAWSHelper) chooseRegion(region string, profile string) (string, error) {
+	suggestion := region
+	if suggestion == "" {
+		region, err := getRegion(profile)
 		if err != nil {
 			return "", err
 		}
-	}
-	reg, err := section.GetKey("region")
-	if err == nil {
-		suggestion = reg.Value()
+		suggestion = region
 	}
 	// promp user for region
-	region, err = h.user.Input("Region", suggestion)
+	region, err := h.user.Input("Region", suggestion)
 	if err != nil {
 		return "", err
 	}
 	if region == "" {
 		return "", fmt.Errorf("region cannot be empty")
 	}
-	// save selected/typed region under profile in ~/.aws/config
-	_, err = section.NewKey("region", region)
-	if err != nil {
-		return "", err
-	}
-	return region, configIni.SaveTo(awsConfig)
+	return region, nil
 }
 
 func (h contextCreateAWSHelper) askCredentials() (string, string, error) {
-	confirm, err := h.user.Confirm("Enter AWS credentials", false)
-	if err != nil {
-		return "", "", err
-	}
-	if !confirm {
-		return "", "", nil
-	}
-
 	accessKeyID, err := h.user.Input("AWS Access Key ID", "")
```

Comment:
Could be helpfull to print link to `https://console.aws.amazon.com/iam/home?#security_credential` so that noobs don't ask ;)